### PR TITLE
Initial commit for enabling GuardDuty in Landing account

### PR DESCRIPTION
**Pull request by obutterbach**
Initial commit for enabling AWS GuardDuty in landing account:
- Create a pipeline for deploying Terraform
- Module GuardDuty Created
- Repo passing kitchen test

### CODE REVIEW

**Comment by djsd123** [comment]
It looks good.  I had a thought though.  Is the implementation in the pipeline module identical to @sh31's? If so @sh31 or yourself should extract it to its own repo. Then you could use it in this PR.  

Also, you have not added any tests(Not that I could see).  We can chat tomorrow.

**Comment by obutterbach** [comment]
Based on our conversation, I agree with you that we should merge this pipeline in 1 place,
Let's discuss about it today 👍

**Comment by obutterbach** [comment]
Updated this PR removing pipeline part, I'll see with Serj for placing this pipeline into a module

File: modules/guardduty/variables.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+variable "region" {
+  default = "eu-west-1"
+}
+
+variable "publish_frequency" {
+  default = "FIFTEEN_MINUTES"
+}
+
+variable "landing_account_id" {}
```
Comment (djsd123) [inline modules/guardduty/variables.tf]: Is it that this module can only enable guard duty on the landing account?  Would we not want guard duty enabled on all our accounts?

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+provider "aws" {
+  region  = "eu-west-1"
+  version = "~> 2.5"
+}
+
+module "aws_guardduty" {
+  source = "modules/guardduty"
+
+  landing_account_id = "${var.landing_account_id}"
```
Comment (djsd123) [inline main.tf]: This variable doesn't seem to exist in `variables.tf`

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    region         = "eu-west-1"
+    dynamodb_table = "tf-state-lock"
+    kms_key_id     = "arn:aws:kms:eu-west-1:335823981503:key/925a5b6c-7df1-49a0-a3cc-471e8524637d"
+  }
+}
+
+provider "aws" {
+  region  = "eu-west-1"
+  version = "~> 2.6"
+}
```
Comment (djsd123) [inline variables.tf:18]: Think you're missing the variable `landing_account_id` in this file.

```
variable "landing_account_id" {
   description = "The id of the landing account"
}
```

File: vars/landing.tfvars [hunk 4]
```diff
@@ -0,0 +1,6 @@
+landing_account_id = "335823981503"
+ap_accounts        = {
+  data = "593291632749"
+  dev  = "525294151996"
+  prod = "312423030077"
+}
```
Comment (djsd123) [inline vars/landing.tfvars]: I'd rename this file to something like __ap_accounts.tfvars__ (your choice).  You could also just add the landing value to the `ap_accounts` map as it has no significance here (I think 🤷‍♂)???

File: modules/guardduty/variables.tf [hunk 5]
```diff
@@ -0,0 +1,19 @@
+terraform {
+  required_version = "~> 0.11.0"
+  backend          "local"          {}
+}
+
+provider "aws" {
```
Comment (djsd123) [inline modules/guardduty/variables.tf]: How does this module exec into an account to enable guard duty?

You might need to assume the roles you created in your other [PR](https://github.com/ministryofjustice/analytical-platform-iam/blob/bef6a1960c4eb46cf89470272c9de7f6502d489d/modules/role/main.tf#L1-L8)?

Example [here](https://github.com/ministryofjustice/analytical-platform-iam/blob/bef6a1960c4eb46cf89470272c9de7f6502d489d/modules/role/main.tf#L1-L8)

File: modules/guardduty/variables.tf (hunk 1 above)
Comment (obutterbach) [reply modules/guardduty/variables.tf]: I'm indeed planning to enable it in all AWS accounts we have
To start with, I'm just enabling it in the landing account, I would then re-iterate for deploying it in different accounts:
- Landing account would send GuardDuty invitations to member accounts
- Member accounts enable and accept GuardDuty invitations

We would then make sure that all GuardDuty events are gather in 1 account only

File: vars/landing.tfvars (hunk 4 above)
Comment (obutterbach) [reply vars/landing.tfvars]: See my reply to your comment (first one) :)

Comment (obutterbach) [reply vars/landing.tfvars]: I'm going to rename it

File: modules/guardduty/variables.tf (hunk 5 above)
Comment (obutterbach) [reply modules/guardduty/variables.tf]: Oh thanks! yeah I see what you mean, let me fix that

**Comment by obutterbach** [comment]
@djsd123 let me know if I capture properly the necessary changes to this PR 😄

File: modules/guardduty/variables.tf [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Assume special IAM-Role (../iam-role) in remote account to create roles and policies
+provider "aws" {
+  region = "${var.region}"
+
+  assume_role {
+    role_arn = "arn:aws:iam::${var.assume_role_in_account_id}:role/${var.landing_iam_role}"
```
Comment (djsd123) [inline modules/guardduty/variables.tf]: Almost mate.  You just need to create the two variables referenced here.  @sh31 should be able to show you what they're doing. 🙏

**Comment by sh31** [comment]
As you're using the existing state bucket and lock table, the state-config directory doesn't need to be here anymore.

**Review by djsd123 — CHANGES_REQUESTED** [review]
Looks good! Once you remove the __state-config__, I think it will be 💯

**Review by djsd123 — APPROVED** [review]
👍 LGTM

**Pull request merged**